### PR TITLE
Entity Status Constants

### PR DESCRIPTION
**Pull request by OroArmor**
Unpicks the constants for entity statuses. Doesnt seem to be working, so help on that would be great

### CODE REVIEW

**Comment by modmuss50** [comment]
Bytes are unsupported in unpick right now, that will be why its not working.

**Comment by OroArmor** [comment]
ah yeah that would be it

**Comment by OroArmor** [comment]
waiting on https://github.com/FabricMC/unpick/pull/5 to be merged

File: mappings/net/minecraft/entity/EntityStatusTypes.mapping [hunk 1]
```diff
@@ -0,0 +1,56 @@
+CLASS net/minecraft/class_6024 net/minecraft/entity/EntityStatusTypes
```
Comment (liach) [inline mappings/net/minecraft/entity/EntityStatusTypes.mapping]: Maybe just `EntityStatuses`? I don't see the need for a `Type` suffix.

Comment (OroArmor) [reply mappings/net/minecraft/entity/EntityStatusTypes.mapping]: Yeah thats better. Ill fix it up one the byte or is merged

Comment (liach) [reply mappings/net/minecraft/entity/EntityStatusTypes.mapping]: Imo we can have this merged in first before the byte support is in. Consider changing the name now. If we delay merging, imo we are just getting more trouble as merge conflicts may come in over time.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/entity/EntityStatuses.mapping [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	FIELD field_29997 BLOCK_WITH_SHIELD B
+	FIELD field_29998 BREAK_SHIELD B
+	FIELD field_29999 PULL_HOOKED_ENTITY B
+	FIELD field_30000 HIT_ARMOR_STAND B
+	FIELD field_30001 DAMAGE_FROM_THORNS B
+	FIELD field_30002 STOP_LOOKING_AT_VILLAGER B
+	FIELD field_30003 USE_TOTEM_OF_UNDYING B
+	FIELD field_30004 DAMAGE_FROM_DROWNING B
+	FIELD field_30005 DAMAGE_FROM_FIRE B
+	FIELD field_30006 ADD_DOLPJIN_HAPPY_VILLAGER_PARTICLES B
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/EntityStatuses.mapping]:
```suggestion
	FIELD field_30006 ADD_DOLPHIN_HAPPY_VILLAGER_PARTICLES B
```

File: unpick-definitions/entity_status.unpick [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+constant entity_statuses net/minecraft/entity/EntityStatuses BLOCK_WITH_SHIELD
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_SHIELD
+constant entity_statuses net/minecraft/entity/EntityStatuses PULL_HOOKED_ENTITY
+constant entity_statuses net/minecraft/entity/EntityStatuses HIT_ARMOR_STAND
+constant entity_statuses net/minecraft/entity/EntityStatuses DAMAGE_FROM_THORNS
+constant entity_statuses net/minecraft/entity/EntityStatuses STOP_LOOKING_AT_VILLAGER
+constant entity_statuses net/minecraft/entity/EntityStatuses USE_TOTEM_OF_UNDYING
+constant entity_statuses net/minecraft/entity/EntityStatuses DAMAGE_FROM_DROWNING
+constant entity_statuses net/minecraft/entity/EntityStatuses DAMAGE_FROM_FIRE
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_DOLPJIN_HAPPY_VILLAGER_PARTICLES
```
Comment (Juuxel) [inline unpick-definitions/entity_status.unpick]:
```suggestion
constant entity_statuses net/minecraft/entity/EntityStatuses ADD_DOLPHIN_HAPPY_VILLAGER_PARTICLES
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/EntityStatuses.mapping [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	FIELD field_30039 ADD_VILLAGER_HAPPY_PARTICLES B
+	FIELD field_30040 ADD_WITCH_PARTICLES B
+	FIELD field_30041 PLAY_CURE_ZOMBIE_VILLAGER_SOUND B
+	FIELD field_30042 EXPLODE_FIREWORK_CLIENT B
+	FIELD field_30043 ADD_BREEDING_PARTICLES B
+	FIELD field_30044 RESET_SQUID_THRUST_TIMER B
+	FIELD field_30045 PLAY_SPAWN_EFFECTS B
+	FIELD field_30046 PLAY_GUARDIAN_ATTACK_SOUND B
+	FIELD field_30048 USE_REDUCED_DEBUG_INFO B
+	FIELD field_30049 DO_NOT_USE_REDUCED_DEBUG_INFO_AND_SET_OP_LEVEL_0 B
```
Comment (enbrain) [inline mappings/net/minecraft/entity/EntityStatuses.mapping]:
```suggestion
	FIELD field_30047 USE_REDUCED_DEBUG_INFO B
	FIELD field_30048 DO_NOT_USE_REDUCED_DEBUG_INFO B
	FIELD field_30049 SET_OP_LEVEL_0 B
```

File: unpick-definitions/entity_status.unpick [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_VILLAGER_HAPPY_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_WITCH_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_CURE_ZOMBIE_VILLAGER_SOUND
+constant entity_statuses net/minecraft/entity/EntityStatuses EXPLODE_FIREWORK_CLIENT
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_BREEDING_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses RESET_SQUID_THRUST_TIMER
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_SPAWN_EFFECTS
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_GUARDIAN_ATTACK_SOUND
+constant entity_statuses net/minecraft/entity/EntityStatuses USE_REDUCED_DEBUG_INFO
+constant entity_statuses net/minecraft/entity/EntityStatuses DO_NOT_USE_REDUCED_DEBUG_INFO_AND_SET_OP_LEVEL_0
```
Comment (enbrain) [inline unpick-definitions/entity_status.unpick]:
```suggestion
constant entity_statuses net/minecraft/entity/EntityStatuses USE_REDUCED_DEBUG_INFO
constant entity_statuses net/minecraft/entity/EntityStatuses DO_NOT_USE_REDUCED_DEBUG_INFO
constant entity_statuses net/minecraft/entity/EntityStatuses SET_OP_LEVEL_0
```

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/EntityStatuses.mapping (hunk 4 above)
Comment (OroArmor) [reply mappings/net/minecraft/entity/EntityStatuses.mapping]: was something messed up?

Comment (OroArmor) [reply mappings/net/minecraft/entity/EntityStatuses.mapping]: oh yeah it looks like it was

File: unpick-definitions/entity_status.unpick [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+constant entity_statuses net/minecraft/entity/EntityStatuses DAMAGE_FROM_BERRY_BUSH
+constant entity_statuses net/minecraft/entity/EntityStatuses CREATE_EATING_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_PORTAL_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_MAINHAND
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_OFFHAND
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_HEAD
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_CHEST
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_LEGS
+constant entity_statuses net/minecraft/entity/EntityStatuses BREAK_FEET
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_SPRINTING_PARTICLES_OR_RESET_SPAWNER_MINECART_SPAWN_DELAY
```
Comment (YanisBft) [inline unpick-definitions/entity_status.unpick:29]: huh

File: mappings/net/minecraft/entity/EntityStatuses.mapping [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	FIELD field_30039 ADD_VILLAGER_HAPPY_PARTICLES B
+	FIELD field_30040 ADD_WITCH_PARTICLES B
+	FIELD field_30041 PLAY_CURE_ZOMBIE_VILLAGER_SOUND B
+	FIELD field_30042 EXPLODE_FIREWORK_CLIENT B
+	FIELD field_30043 ADD_BREEDING_PARTICLES B
+	FIELD field_30044 RESET_SQUID_THRUST_TIMER B
+	FIELD field_30045 PLAY_SPAWN_EFFECTS B
+	FIELD field_30046 PLAY_GUARDIAN_ATTACK_SOUND B
+	FIELD field_30047 USE_REDUCED_DEBUG_INFO B
+	FIELD field_30048 DO_NOT_USE_REDUCED_DEBUG_INFO B
```
Comment (enbrain) [inline mappings/net/minecraft/entity/EntityStatuses.mapping]:
```suggestion
	FIELD field_30048 USE_FULL_DEBUG_INFO B
```

File: unpick-definitions/entity_status.unpick [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_VILLAGER_HAPPY_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_WITCH_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_CURE_ZOMBIE_VILLAGER_SOUND
+constant entity_statuses net/minecraft/entity/EntityStatuses EXPLODE_FIREWORK_CLIENT
+constant entity_statuses net/minecraft/entity/EntityStatuses ADD_BREEDING_PARTICLES
+constant entity_statuses net/minecraft/entity/EntityStatuses RESET_SQUID_THRUST_TIMER
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_SPAWN_EFFECTS
+constant entity_statuses net/minecraft/entity/EntityStatuses PLAY_GUARDIAN_ATTACK_SOUND
+constant entity_statuses net/minecraft/entity/EntityStatuses USE_REDUCED_DEBUG_INFO
+constant entity_statuses net/minecraft/entity/EntityStatuses DO_NOT_USE_REDUCED_DEBUG_INFO
```
Comment (enbrain) [inline unpick-definitions/entity_status.unpick]:
```suggestion
constant entity_statuses net/minecraft/entity/EntityStatuses USE_FULL_DEBUG_INFO
```

**Review by enbrain — COMMENTED** [review]
Looks good besides one nitpick.

File: unpick-definitions/entity_status.unpick (hunk 6 above)
Comment (liach) [reply unpick-definitions/entity_status.unpick:29]: Imo this is good. We don't know if they could have a separate field for this minecart delay, but having this here imo is good enough

File: mappings/net/minecraft/entity/EntityStatuses.mapping (hunk 7 above)
Comment (OroArmor) [reply mappings/net/minecraft/entity/EntityStatuses.mapping]: yeah this is good

File: unpick-definitions/entity_status.unpick (hunk 6 above)
Comment (OroArmor) [reply unpick-definitions/entity_status.unpick:29]: There is. its `field_30951`, but it is `private static final`.

**Review by enbrain — APPROVED** [review]

File: unpick-definitions/entity_status.unpick (hunk 6 above)
Comment (OroArmor) [reply unpick-definitions/entity_status.unpick:29]: I think its fine if the constants file is a little messy if it covers all of the cases. I don't think its going to confuse too many people. also mojang should have things overlap, so we work with what we have

Comment (liach) [reply unpick-definitions/entity_status.unpick:29]: >  but it is `private static final`.

Hmm, is there a public alternative then? Something must be using these values from fields elsewhere.

Comment (OroArmor) [reply unpick-definitions/entity_status.unpick:29]: no not that i saw. its only referenced once in `MobSpawnerLogic#handleStatus`

Comment (liach) [reply unpick-definitions/entity_status.unpick:29]: Hmm, in fact this is the 1st field in the class and has value `1`. Dunno how stitch assign number to the fields; it is just weird.

**Review by YanisBft — APPROVED** [review]

**Comment by OroArmor** [comment]
Uh the unpick stuff might not work, so if you only merged in the mappings for now, then once https://github.com/FabricMC/unpick/pull/5 is merged, i make the unpick pr

**Pull request merged**